src/main/java/org/spongepowered/api/map/color/MapColor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.color;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.util.Color;
+
+public interface MapColor extends DataSerializable {
+
+    /**
+     * Returns the corresponding {@link Color} for this {@link MapColor}.
+     *
+     * @return The RGB color of this map color
+     */
+    Color getColor();
+
+    /**
+     * Creates a shaded representation of this color's <b>base color</b> using
+     * {@link MapPalette#getShade(MapColor, MapShade)} which will create a new
+     * instance if needed or return a cached instance for previously created
+     * shades.
+     *
+     * @param newShading The shading to use on the new variant of this color
+     * @return The new map color with the specified shading
+     */
+    MapColor createShade(MapShade newShading);
```

Comment:
No. MapCanvas represents the pixel array and provides operations to modify
it. The MapView represents the whole map data. Combining the two would be
possible yes, but not as clear.

On Sat, Apr 2, 2016, 3:37 PM Starbuck Johnson <starbuck@modwiz.com> wrote:

> Indeed. That is a good point, I believe a MapPixel and a MapColor should
> be available. In fact the concept of MapPixel solves another issue I've
> been considering which is setting blocks of pixels. I will be working this
> into the PR in a few hours.
>
> On Sat, Apr 2, 2016, 3:30 PM Gabriel Harris-Rouquette <
> notifications@github.com> wrote:
>
>> In src/main/java/org/spongepowered/api/map/color/MapColor.java
>> <https://github.com/SpongePowered/SpongeAPI/pull/1162#discussion_r58299452>
>> :
>>
>> > +     * Returns the corresponding {@link Color} for this {@link MapColor}.
>> > +     *
>> > +     * @return The RGB color of this map color
>> > +     */
>> > +    Color getColor();
>> > +
>> > +    /**
>> > +     * Creates a shaded representation of this color's <b>base color</b> using
>> > +     * {@link MapPalette#getShade(MapColor, MapShade)} which will create a new
>> > +     * instance if needed or return a cached instance for previously created
>> > +     * shades.
>> > +     *
>> > +     * @param newShading The shading to use on the new variant of this color
>> > +     * @return The new map color with the specified shading
>> > +     */
>> > +    MapColor createShade(MapShade newShading);
>>
>> This isn't creating a shade, this is creating a new MapColor with the
>> provided MapShade, is it not? The method should be renamed.
>>
>> —
>> You are receiving this because you were mentioned.
>>
>> Reply to this email directly or view it on GitHub
>> <https://github.com/SpongePowered/SpongeAPI/pull/1162/files/d8338269be828e278d15ed5ffbe9bb30b7c899c1#r58299452>
>>
>